### ISSUE TITLE
[feature request] Use a custom API endpoint

### ISSUE THREAD

**Issue by josuah** [brilliantlabsAR/noa-assistant#31]
Currently, only the OpenAI ChatGPT and PerplexityAI services are supported by Noa.

Some users run their own private server, and are interested in adding their own API endpoint URL.

**Comment by josuah** [comment]
Requested from https://discord.com/channels/963222352534048818/984966420603482182/1267964841084059793

**Comment by wbste** [comment]
Yeah if you could support any openai compatible endpoints this would be a huge benefit. LM Studio, Ollama, llama.cpp, etc. could be used for fully private AI.
